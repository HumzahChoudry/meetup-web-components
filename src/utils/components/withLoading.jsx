import PropTypes from 'prop-types';
import React from 'react';

import Loading from '../../media/Loading';

const withLoading = (WrappedComponent) => {
	/**
	 * @module WithLoading
	 */
	class WithLoading extends React.Component {
		render() {
			const {
				children,
				isLoading,
				loadingProps,
				...other
			} = this.props;

			return(
				<WrappedComponent
					aria-busy={isLoading}
					isLoading={isLoading}
					/* loadingComponent={
						isLoading &&
							<Loading
								partialCover
								{...loadingProps}
							/>
					}*/
					{...other}
				>
					{this.props.dangerouslySetInnerHTML
						? null
						: [children, isLoading && <Loading key="loadingIndicator" partialCover {...loadingProps}/>]}
				</WrappedComponent>
			);
		}
	}

	return WithLoading;
};

withLoading.defaultProps = {
	isLoading: false
};
withLoading.propTypes = {
	isLoading: PropTypes.bool,
	loadingProps: PropTypes.shape({
		color: PropTypes.string,
		scrimColor: PropTypes.string,
		size: PropTypes.string
	})
};

export default withLoading;
