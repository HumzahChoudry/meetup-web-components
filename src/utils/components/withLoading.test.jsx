import React from 'react';
import { shallow } from 'enzyme';

import withLoading from './withLoading';

const dangerousHTML = '<div>Dangerously setting inner HTML</div>';
const makeDanger = () => ({__html: dangerousHTML});

const testComponentClass = 'testClass';
const TestComponent = (props) => (<div {...props} />);
const TestComponentWithLoading = withLoading(TestComponent);

describe('withLoading', function() {
	const testComponentWithLoading = shallow(<TestComponentWithLoading className={testComponentClass} />);

	it('matches snapshot', () => {
		expect(testComponentWithLoading).toMatchSnapshot();
	});

	it('renders a wrapped component', () => {
		expect(testComponentWithLoading.find(`.${testComponentClass}`).length).toBe(1);
	});

	it('can handle dangerouslySetInnerHTML', function() {
		const dangerousHTMLComponent = shallow(
			<TestComponentWithLoading
				dangerouslySetInnerHTML={makeDanger()}
			/>
		);
		expect(dangerousHTMLComponent.html()).toContain(dangerousHTML);
	});
});
