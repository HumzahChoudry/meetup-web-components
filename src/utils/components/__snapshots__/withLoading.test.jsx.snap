// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoading matches snapshot 1`] = `
<TestComponent
  className="testClass"
/>
`;
